system/service/apigw_filter.go

```diff
@@ -233,14 +225,14 @@ func (svc *apigwFilter) UndeleteByID(ctx context.Context, ID uint64) (err error)
 			return ApigwFilterErrNotFound(qProps)
 		}
 
-		if !svc.ac.CanDeleteApigwFilter(ctx, q) {
-			return ApigwFilterErrNotAllowedToDelete(qProps)
-		}
-
 		if r, err = svc.route.FindByID(ctx, q.Route); err != nil {
 			return err
 		}
 
+		if !svc.ac.CanDeleteApigwRoute(ctx, r) {
+			return ApigwRouteErrNotAllowedToUndelete()
+		}
```

Comment:
The undelete function should check if user can **update** the route, not if they can delete it.